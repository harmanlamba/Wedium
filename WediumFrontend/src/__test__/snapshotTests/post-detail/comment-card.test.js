import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import CommentCard, {
  styles,
} from '../../../components/post-detail/comment/comment-card';

it('snapshot commentCard default load', () => {
  const user = {
    firstName: 'null',
    isAuthenticated: false,
    jwtToken: 'null',
    lastName: 'null',
    pictureUri: 'null',
    username: 'null',
  };

  const comment = {
    body: 'This is a comment',
    commentId: 1,
    commentTypeId: 1,
    date: '2020-05-21T21:56:36.933',
    inverseParentComment: [],
    parentCommentId: null,
    postId: 123,
    userId: 123,
    userName: 'Bob Shaw',
  };

  const setRouteLeaveHook = jest.fn();

  let wrapper = shallow(
    <CommentCard.WrappedComponent
      user={user}
      comment={comment}
      classes={styles}
      params={{
        router: setRouteLeaveHook,
      }}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('snapshot postCommentFeed with user authenticated load', () => {
  const user = {
    firstName: 'Bob',
    isAuthenticated: true,
    jwtToken: 'jwtToken',
    lastName: 'Jenkins',
    pictureUri: 'somePhoto',
    username: 'Bob Jenkins',
  };

  const comment = {
    body: 'This is a comment',
    commentId: 1,
    commentTypeId: 1,
    date: '2020-05-21T21:56:36.933',
    inverseParentComment: [],
    parentCommentId: null,
    postId: 123,
    userId: 123,
    userName: 'Bob Shaw',
  };

  const setRouteLeaveHook = jest.fn();

  let wrapper = shallow(
    <CommentCard.WrappedComponent
      user={user}
      comment={comment}
      classes={styles}
      params={{
        router: setRouteLeaveHook,
      }}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
