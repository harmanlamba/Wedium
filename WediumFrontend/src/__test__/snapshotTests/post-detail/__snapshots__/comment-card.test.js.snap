// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot commentCard default load 1`] = `
<div
  className=""
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    justify="flex-end"
    spacing={0}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(Connect(CommentLike))
        commentId={1}
        parentCommentId={null}
        user={
          Object {
            "firstName": "null",
            "isAuthenticated": false,
            "jwtToken": "null",
            "lastName": "null",
            "pictureUri": "null",
            "username": "null",
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={10}
    >
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        variant="caption"
      >
        Bob Shaw
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        variant="caption"
      >
        Invalid date
         -
         
        Invalid date
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={10}
    >
      This is a comment
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Collapse))
      in={false}
      timeout="auto"
      unmountOnExit={true}
    >
      <div>
        <WithStyles(RichTextBox)
          isEmptyNow={false}
          maxLength={350}
          onChange={[Function]}
          placeholder="Write a reply here..."
          setIsEmptyNow={[Function]}
        />
        <WithStyles(ForwardRef(Button))
          color="primary"
          onClick={[Function]}
          size="small"
          variant="contained"
        >
          Reply
        </WithStyles(ForwardRef(Button))>
      </div>
    </WithStyles(ForwardRef(Collapse))>
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={2000}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(Alert))
      onClose={[Function]}
      severity="warning"
    >
      Reply is empty!
    </WithStyles(ForwardRef(Alert))>
  </WithStyles(ForwardRef(Snackbar))>
  <WithStyles(ForwardRef(CardActions))
    className=""
    disableSpacing={true}
  />
  <WithStyles(ForwardRef(Collapse))
    in={false}
    timeout="auto"
    unmountOnExit={true}
  />
</div>
`;

exports[`snapshot postCommentFeed with user authenticated load 1`] = `
<div
  className=""
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    justify="flex-end"
    spacing={0}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(IconButton))
        aria-label="Comment button"
        onClick={[Function]}
        size="small"
      >
        <ChatBubbleOutlineIcon
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(Connect(CommentLike))
        commentId={1}
        parentCommentId={null}
        user={
          Object {
            "firstName": "Bob",
            "isAuthenticated": true,
            "jwtToken": "jwtToken",
            "lastName": "Jenkins",
            "pictureUri": "somePhoto",
            "username": "Bob Jenkins",
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={10}
    >
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        variant="caption"
      >
        Bob Shaw
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        variant="caption"
      >
        Invalid date
         -
         
        Invalid date
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={10}
    >
      This is a comment
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Collapse))
      in={false}
      timeout="auto"
      unmountOnExit={true}
    >
      <div>
        <WithStyles(RichTextBox)
          isEmptyNow={false}
          maxLength={350}
          onChange={[Function]}
          placeholder="Write a reply here..."
          setIsEmptyNow={[Function]}
        />
        <WithStyles(ForwardRef(Button))
          color="primary"
          onClick={[Function]}
          size="small"
          variant="contained"
        >
          Reply
        </WithStyles(ForwardRef(Button))>
      </div>
    </WithStyles(ForwardRef(Collapse))>
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={2000}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(Alert))
      onClose={[Function]}
      severity="warning"
    >
      Reply is empty!
    </WithStyles(ForwardRef(Alert))>
  </WithStyles(ForwardRef(Snackbar))>
  <WithStyles(ForwardRef(CardActions))
    className=""
    disableSpacing={true}
  />
  <WithStyles(ForwardRef(Collapse))
    in={false}
    timeout="auto"
    unmountOnExit={true}
  />
</div>
`;
